bulk_indexer.go

```diff
@@ -501,27 +492,11 @@ func (b *BulkIndexer) Flush(ctx context.Context) (BulkIndexerResponseStat, error
 	b.bytesUncompFlushed = bytesUncompFlushed
 	var resp BulkIndexerResponseStat
 	if res.StatusCode > 299 {
-		var s string
-		var er struct {
-			Error struct {
-				Type     string `json:"type,omitempty"`
-				Reason   string `json:"reason,omitempty"`
-				CausedBy struct {
-					Type   string `json:"type,omitempty"`
-					Reason string `json:"reason,omitempty"`
-				} `json:"caused_by,omitempty"`
-			} `json:"error,omitempty"`
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			return BulkIndexerResponseStat{}, fmt.Errorf("failed to read response body: %w", err)
 		}
-
-		if err := jsoniter.NewDecoder(res.Body).Decode(&er); err == nil {
-			er.Error.Reason = ""
-			er.Error.CausedBy.Reason = ""
-
-			b, _ := json.Marshal(&er)
-			s = string(b)
-		}
-
-		e := errorFlushFailed{resp: s, statusCode: res.StatusCode}
+		e := errorFlushFailed{resp: string(b), statusCode: res.StatusCode}
```

Comment:
if we're going with plan (2) where we remove sanitization and offload it to docappender users, the way `resp: string(b)` is passed and printed into a string in `errorFlushFailed.Error()` makes it very hard for docappender users to parse.